fendermint/app/src/main.rs

```diff
@@ -30,6 +31,88 @@ async fn main() {
     }
 }
 
+fn create_log(level: tracing::Level, log_dir: Option<&PathBuf>) -> anyhow::Result<()> {
+    let console_appender = config::Appender::Console;
+
+    // default logging output info log to console
+    let default = config::Logger {
+        level: config::LevelFilter::from_str(level.as_str())?,
+        appenders: literally::hset! { AppenderId::from("console") },
+        format: Format::default(),
+    };
+
+    let mut config = Config {
+        default,
+        loggers: literally::hmap! {},
+        appenders: literally::hmap! {AppenderId::from("console") => console_appender},
+    };
+
+    if let Some(log_dir) = log_dir {
+        let log_folder = expand_tilde(log_dir)
+            .to_str()
+            .ok_or_else(|| anyhow!("cannot parse log folder"))?
+            .to_string();
+        std::fs::create_dir_all(&log_folder).context("cannot create log folder")?;
+
+        let topdown_appender = config::Appender::RollingFile {
+            path: format!("{log_folder}/topdown.log"),
+            policy: Policy {
+                maximum_file_size: "10mb".to_string(),
+                // we keep the last 5 log files, older files will be deleted
+                max_size_roll_backups: 5,
+                pattern: None,
+            },
+        };
+        let debug_appender = config::Appender::RollingFile {
+            path: format!("{log_folder}/debug.log"),
+            policy: Policy {
+                maximum_file_size: "10mb".to_string(),
+                // we keep the last 5 log files, older files will be deleted
+                max_size_roll_backups: 5,
+                pattern: None,
+            },
+        };
+
+        let topdown_logger = config::Logger {
+            level: config::LevelFilter::DEBUG,
+            appenders: literally::hset! { AppenderId::from("topdown") },
+            format: Format::default(),
+        };
+        let debug_logger = config::Logger {
+            level: config::LevelFilter::DEBUG,
+            appenders: literally::hset! { AppenderId::from("debug") },
+            format: Format::default(),
+        };
+
+        config.loggers.insert(
+            Target::from("fendermint_vm_topdown"),
+            topdown_logger.clone(),
+        );
+        config.loggers.insert(
+            Target::from("fendermint_vm_interpreter::chain"),
+            topdown_logger,
+        );
+        config
+            .appenders
+            .insert(AppenderId::from("topdown"), topdown_appender);
+
+        // maybe there are better ways to handle this, but * does not seem to be supported
+        config
+            .loggers
+            .insert(Target::from("fendermint"), debug_logger);
+        config
+            .appenders
+            .insert(AppenderId::from("debug"), debug_appender);
```

Comment:
Yeah so why don't we insert a clone of the `default` console logger with an `Target::from("fendermint")` and the `INFO` level?

Reply:
yeah, but it seems to me this is a bug or sth, we should not be able to do that though. But I added this.

Reply:
What's a bug? Why shouldn't we be able to set a logger that captures everything and logs to the console if we want to?

Reply:
I mean we need to do this:
```rust
        config
            .loggers
            .insert(Target::from("fendermint"), debug_logger);
        config.loggers.insert(Target::from("fendermint"), default);
```
but ideally we should not need `config.loggers.insert(Target::from("fendermint"), default);` as we already set default.